### ISSUE TITLE
Suggestion to remove Address type

### ISSUE THREAD

**Issue by kevaundray** [AztecProtocol/barretenberg#376]
## Observation

The stdlib has an address type, which can be seen [here](https://github.com/AztecProtocol/barretenberg/blob/ecb61292c96c3b1fc673bcd96920cd2f00fe28b9/cpp/src/barretenberg/stdlib/primitives/address/address.hpp#L23)

## Comments

I'm assuming this is meant to be an Aztec 3 address, if so, this is scope creep as a cryptography library should not need to know what an Aztec 3 address is or the concept of a blockchain address, since an address is a protocol specific primitive that has nothing to do with the cryptography being done in barretenberg.

## Suggestion

Move address into the downstream package that requires this concept

**Comment by codygunton** [comment]
Agree, writing code risks changes in Barretenberg breaking downstream Aztec work. But more generally, it just doesn't make sense to have this type in the backend library that is ostensibly for use outside of Aztec Labs.

**Comment by codygunton** [comment]
Also: why do initialize from `int`s there? Signed types make me nervous.
